cart/test/api.test.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-env mocha */
+const chai = require("chai");
+const { expect } = chai;
+const host = "localhost";
+const port = 3000;
+const cartURL = `http://${host}:${port}/cart`; // URL for GraphQL API Gateway
+const requestCart = require("supertest")(cartURL);
+const logger = require('../src/utils/logger');
+
+describe("Cart REST API", () => {
+  let product, cartid, cart, sample_products, sample_users;
+
+  // Utility function to initialize data
+  async function loadSampleData() {
+    // Set up test data
+    sample_users = [
+      {
+        "uid": "user1",
+        "fname": "John",
+        "lname": "Doe",
+        "address": "Some Street 22",
+        "phone": 1231231231,
+        "email": "john@doe.com"
+      }
+    ]
+
+    sample_products = [
+      {
+        "pid": 1,
+        "pcode": "ABC123",
+        "price": 42.5,
+        "sku": "XYZ",
+        "amount_in_stock": 10,
+        "pname": "Something",
+        "description": "This is something very interesting that you want to buy.",
+        "lang": "en_US"
+      },
+      {
+        "pid": 2,
+        "pcode": "FD2",
+        "price": 99.99,
+        "sku": "QWOP",
+        "amount_in_stock": 100,
+        "pname": "Speedos",
+        "description": "I'm too lazy to write a description",
+        "lang": "en_US"
+      }
+    ]
+
+    // Create the records in the database
+    logger.debug(`Inserting sample records`);
```

Comment:
no need to log when running tests. use console log instead... this is a Test_env and logs in test_env should not be logged to a file. Also it doesn't add much value if logged and will feed logs with unwanted info

Reply:
It doesn't actually go into the application log, so there's no difference. But I see how that could introduce confusion, so I changed it.